pandas/tseries/tests/test_tslib.py

```diff
@@ -180,6 +180,36 @@ def test_constructor_invalid(self):
         with tm.assertRaisesRegexp(ValueError, 'Cannot convert Period'):
             Timestamp(Period('1000-01-01'))
 
+    def test_constructor_positional(self):
+        # GH 10758
+        with tm.assertRaises(TypeError):
+            Timestamp(2000, 1)
+        with tm.assertRaises(ValueError):
+            Timestamp(2000, 0, 1)
+        with tm.assertRaises(ValueError):
+            Timestamp(2000, 13, 1)
+        with tm.assertRaises(ValueError):
+            Timestamp(2000, 1, 0)
+        with tm.assertRaises(ValueError):
+            Timestamp(2000, 1, 32)
+        with tm.assertRaises(TypeError):
+            Timestamp(2000, 1, 1, None)
+        with tm.assertRaises(TypeError):
+            Timestamp(2000, 1, 1, None, None)
+        with tm.assertRaises(TypeError):
+            Timestamp(2000, 1, 1, None, None, None)
+
+        ts = Timestamp(2000, 1, 2)
+
+        actual = ts.to_pydatetime()
+        expected = datetime.datetime(2000, 1, 2)
+        self.assertEqual(actual, expected)
+        self.assertEqual(type(actual), type(expected))
+
+        pos_args = [2000, 1, 2, 3, 4, 5, 999999]
```

Comment:
can u also add another with ``tzinfo`` (another set of pos_args).

and another with kwargs as well.